new-client/src/plugins/Collector/components/Toolbar.js

```diff
@@ -37,6 +38,24 @@ class Toolbar extends Component {
         activeTool: undefined,
       });
     });
+
+    // Clear layer and attempt to read saved values
+    if (this.props.model.wkt) {
+      this.props.model.vectorSource.clear();
+
+      try {
+        const format = new WKT();
+        let features = format.readFeaturesFromText(
+          this.props.model.formValues[this.props.field]
+        );
+        features.forEach((feature) => {
+          this.props.model.vectorSource.addFeature(feature);
+        });
```

Comment:
The readFeaturesFromText is a protected method and should not be used i think. See, row 625 in new-client/node_modules/ol/format/WKT.js. Maybey there is a better method to use? If not, i will still accept the change since this is a stand-alone plugin. 

I think we can avoid the try-catch by making sure features is an array of features (or empty) before trying to map over it?

Reply:
Now using readFeatures instead and restructured to avoid the try-catch as well.